File: 3D Multiplayer GP2/Assets/Modern Day Characters/Base Male/Scripts/C#/GUI/ButtonSetMaterial.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class ButtonSetMaterial : MonoBehaviour {
	public Material[] pickMaterials;
	public int currentMaterialID = 0;
	public SkinnedMeshRenderer chooseMeshRenderer;

	void Update() {
		if(Input.GetMouseButtonDown(0)) {
			if(GetComponent<GUITexture>().HitTest(Input.mousePosition)) {
				currentMaterialID++;

				if(currentMaterialID >= pickMaterials.Length) {
					currentMaterialID = 0;
				}

				chooseMeshRenderer.material = pickMaterials[currentMaterialID];
			}
		}
	}
}

File: 3D Multiplayer GP2/Assets/Modern Day Characters/Base Male/Scripts/JavaScript/GUI/ButtonSetMaterial.js
#pragma strict

public var pickMaterials : Material[];
public var currentMaterialID : int = 0;
public var chooseMeshRenderer : SkinnedMeshRenderer;

function Update() {
	if(Input.GetMouseButtonDown(0)) {
		if(GetComponent.<GUITexture>().HitTest(Input.mousePosition)) {
			currentMaterialID ++;

			if(currentMaterialID >= pickMaterials.Length) {
				currentMaterialID = 0;
			}

			chooseMeshRenderer.material = pickMaterials[currentMaterialID];
		}
	}
}